Point_set_processing_3/include/CGAL/structure_point_set.h

```diff
@@ -0,0 +1,1483 @@
+                      b1 = true;
+                  }
+                if (a0 && a1 && b0 && b1)
+                  return VERTEX;
+                else
+                  return INCOHERENT;
+              }
+            else if (sa == PLANE && sb == PLANE)
+              {
+                if (a != b)
+                  return INCOHERENT;
+
+                for (std::size_t j = 0; j < m_corners[c].planes.size (); ++ j)
+                  if (m_corners[c].planes[j] == a)
+                    return VERTEX;
+                  
+                return INCOHERENT;
+              }
+            else if (sa == PLANE && sb == EDGE)
+              {
+                bool pa = false, b0 = false, b1 = false;
+                if (a != m_edges[b].planes[0] && a != m_edges[b].planes[1])
+                  return INCOHERENT;
+                  
+                for (std::size_t j = 0; j < m_corners[c].planes.size (); ++ j)
+                  {
+                    if (m_corners[c].planes[j] == a)
+                      pa = true;
+                    if (m_corners[c].planes[j] == m_edges[b].planes[0])
+                      b0 = true;
+                    else if (m_corners[c].planes[j] == m_edges[b].planes[1])
+                      b1 = true;
+                  }
+                if (pa && b0 && b1)
+                  return VERTEX;
+                else
+                  return INCOHERENT;
+              }
+            else if (sa == EDGE && sb == PLANE)
+              {
+                bool a0 = false, a1 = false, pb = false;
+                if (b != m_edges[a].planes[0] && b != m_edges[a].planes[1])
+                  return INCOHERENT;
+                  
+                for (std::size_t j = 0; j < m_corners[c].planes.size (); ++ j)
+                  {
+                    if (m_corners[c].planes[j] == b)
+                      pb = true;
+                    if (m_corners[c].planes[j] == m_edges[a].planes[0])
+                      a0 = true;
+                    else if (m_corners[c].planes[j] == m_edges[a].planes[1])
+                      a1 = true;
+                  }
+                if (a0 && a1 && pb)
+                  return VERTEX;
+                else
+                  return INCOHERENT;
+              }
+            else
+              return INCOHERENT;
+          }
+      }
+
+
+    return INCOHERENT;
+  }
+
+
+  /// \cond SKIP_IN_MANUAL  
+private:
+
+
+  void clean ()
+  {
+    std::vector<Point> points;
+    std::vector<Vector> normals;
+    std::vector<std::size_t> indices;
+    std::vector<Point_status> status;
+      
+    for (std::size_t i = 0; i < m_points.size (); ++ i)
+      if (m_status[i] != SKIPPED)
+        {
+          points.push_back (m_points[i]);
+          normals.push_back (m_normals[i]);
+          status.push_back (m_status[i]);
+          if (m_status[i] == RESIDUS)
+            status.back () = PLANE;
+          indices.push_back (m_indices[i]);
+        }
+      
```

Comment:
Would it really be worth it in terms of computation time?

Otherwise, I could just reserve with the current size of `m_points`: it's bigger, but that means no reallocation.